### ISSUE TITLE
Support for metadata over IPv6

### ISSUE THREAD

**Issue by ubuntu-server-builder** [canonical/cloud-init#3816]
This bug was originally filed in Launchpad as [LP: #1906849](https://bugs.launchpad.net/cloud-init/+bug/1906849)
<details>
<summary>Launchpad details</summary>
<pre>
affected_projects = []
assignee = None
assignee_name = None
date_closed = 2023-02-22T16:52:18.219741+00:00
date_created = 2020-12-04T15:31:49.599450+00:00
date_fix_committed = 2022-11-20T07:27:09.400476+00:00
date_fix_released = 2023-02-22T16:52:18.219741+00:00
id = 1906849
importance = wishlist
is_complete = True
lp_url = https://bugs.launchpad.net/cloud-init/+bug/1906849
milestone = None
owner = slaweq
owner_name = Slawek Kaplonski
private = False
status = fix_released
submitter = slaweq
submitter_name = Slawek Kaplonski
tags = []
duplicates = []
</pre>
</details>

_Launchpad user **Slawek Kaplonski(slaweq)** wrote on 2020-12-04T15:31:49.599450+00:00_

OpenStack Neutron recently added support to Metadata over IPv6. See spec https://specs.openstack.org/openstack/neutron-specs/specs/ussuri/metadata-add-ipv6-support.html

It would be great if cloud-init would add support for that too. I'm not cloud-init expert at all and I have no idea how it could be implemented but I think it's good to at least propose that here so maybe someone will want to work on that.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Lucas Albuquerque Medeiros de Moura(lamoura)** wrote on 2020-12-07T13:55:28.462874+00:00_

Thanks for filling this feature request

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Maximilian Stinsky(mstinsky)** wrote on 2022-09-19T12:39:27.327765+00:00_

Greetings,

I just stumbled upon this feature request, as we upgraded our openstack setup to a version which supports metadata over ipv6. While testing it we found that cloud-init does not support this without manually setting the metadata_url over user-data.

Maybe I am too naive, but this does not look to be the hardest implementation if it would be done like in ec2 with a metadata url list which is just tested one by one [1].

Could this be done the same way in openstack where we change [2] to be a list of possible metadata addresses?

Any opinions on this?


[1] https://github.com/canonical/cloud-init/blob/8776a661a9189cbc0c4437562bcd3b6c27f81854/cloudinit/sources/DataSourceEc2.py#L66

[2] https://github.com/canonical/cloud-init/blob/919e22dc1d779f58f03dfd1cfde5ad9dc33a60d1/cloudinit/sources/DataSourceOpenStack.py#L21
